app/src/operations_endpoints.h

```diff
@@ -0,0 +1,681 @@
+     * serve as a convenient debugging endpoint into the state of the indexing
+     * strategy.
+     */
+    std::vector<GetOperation::Out> operations() const
+    {
+      std::lock_guard guard(lock);
+      std::vector<GetOperation::Out> result;
+      for (const auto& it : operations_)
+      {
+        result.push_back(GetOperation::Out{
+          .operation_id = ccf::TxID{it.second.view, it.first},
+          .status = it.second.status,
+          .entry_id = it.second.completion_tx,
+          .error = it.second.error,
+        });
+      }
+      return result;
+    }
+
+    /**
+     * Look up an operation by its transaction ID.
+     *
+     * If the transaction ID is unknown, this function generally returns an
+     * operation in the "running" state anyway, since it would not be possible
+     * to distinguish between whether this is an invalid transaction ID or if
+     * the transaction that created the operation has not been globally
+     * committed and indexed yet.
+     *
+     * In some cases however, we can be confident that this transaction ID is
+     * invalid now and forever in the future. In these cases, an appropriate
+     * HTTPError exception is thrown.
+     */
+    GetOperation::Out lookup(const ccf::TxID& operation_id) const
+    {
+      std::lock_guard guard(lock);
+
+      ccf::TxStatus tx_status;
+      auto result =
+        get_status_for_txid(operation_id.view, operation_id.seqno, tx_status);
+      if (result != ccf::ApiResult::OK)
+      {
+        throw InternalError(fmt::format(
+          "Failed to get transaction status: {}",
+          ccf::api_result_to_str(result)));
+      }
+
+      switch (tx_status)
+      {
+        case ccf::TxStatus::Unknown:
+        case ccf::TxStatus::Pending:
+          return {
+            .operation_id = operation_id,
+            .status = OperationStatus::Running,
+          };
+
+        case ccf::TxStatus::Invalid:
+          // This state can arise even in a well-behaved client if the view
+          // changed (eg. because of a Raft election), and the operation's
+          // transaction got dropped. It could also be a client giving us
+          // garbage txids, but we can't tell the difference so we remain polite
+          // and assume the former and say the operation has failed.
+          return {
+            .operation_id = operation_id,
+            .status = OperationStatus::Failed,
+            .error =
+              ODataError{
+                .code = ccf::errors::TransactionInvalid,
+                .message = "Transaction is invalid",
+              },
+          };
+
+        case ccf::TxStatus::Committed:
+          // This is the main case, when the client is referring to a
+          // transaction that has been committed to the ledger. From now on, it
+          // is safe to consider the SeqNo only and compare it to what has been
+          // indexed.
+          break;
+      }
+
+      if (operation_id.seqno < lower_bound)
```

Comment:
This would also be returned when the out-of-range transaction wasn't an operation transaction. I think to keep it simple it's better to just return 404 like below.

Reply:
This code generally assumes the client is well-behaved and is only using actual operations, so it gives it the benefit of the doubt and replies accordingly. This is why this returns a 410 rather than 404.

Only when we know for certain this transaction couldn't have been an operation do we return a 404.